examples/RNOneSignal/App.js

```diff
@@ -8,484 +8,700 @@
-      <ScrollView style={styles.scrollView}>
-        <View style={styles.container}>
-          <View>
-            <Image style={styles.imageStyle} source={{uri: imageUri}} />
-          </View>
-          <Text style={styles.welcome}>Welcome to React Native!</Text>
-          <Text style={styles.instructions}>To get started, edit index.js</Text>
-          <Text style={styles.instructions}>
-            Double tap R on your keyboard to reload,{'\n'}
-            Shake or press menu button for dev menu
-          </Text>
-          <Text style={styles.jsonDebugLabelText}>
-            {this.state.jsonDebugText}
-          </Text>
-          <View style={{flexDirection: 'row', overflow: 'hidden'}}>
-            <View style={styles.buttonContainer}>
-              <Button
-                style={styles.button}
-                onPress={() => {
-                  OneSignal.getTags(tags => {
-                    console.log('Did get tags: ', tags);
-
-                    this.setState({
-                      jsonDebugText: JSON.stringify(tags, null, 2),
-                    });
-                  });
-                }}
-                title="Get Tags"
-                color={this.state.buttonColor}
-              />
+
+    constructor(properties) {
+        super(properties);
+
+        this.state = {
+            // OneSignal states
+            // Privacy Consent states
+            hasPrivacyConsent: false, // App starts without privacy consent
+            isPrivacyConsentLoading: requiresPrivacyConsent,
+
+            // Device states
+            userId: '',
+            pushToken: '',
+
+            // Subscription states
+            isSubscribed: false,
+            isSubscriptionLoading: false,
+
+            // External User Id states
+            externalUserId: '',
+            isExternalUserIdLoading: false,
+
+            // Email states
+            email: '',
+            isEmailLoading: false,
+
+            // In-App Messaging states
+            iam_paused: false,
+
+            // Add more states here...
+
+            // Demo App states
+            debugText: ''
+        };
+
+        OneSignal.setRequiresUserPrivacyConsent(requiresPrivacyConsent);
+        OneSignal.init(appId, {
+            kOSSettingsKeyAutoPrompt: true,
+        });
+
+        // Log level logcat is 6 (VERBOSE) and log level alert is 0 (NONE)
+        OneSignal.setLogLevel(6, 0);
+
+        // Share location of device
+        OneSignal.setLocationShared(true);
+
+        // Notifications will display as NOTIFICATION type
+        OneSignal.inFocusDisplaying(2);
+
+        // If the app requires privacy consent check if it has been set yet
+        if (requiresPrivacyConsent) {
+        // async 'then' is only so I can sleep using the Promise helper method
+            OneSignal.userProvidedPrivacyConsent().then(async (granted) => {
+                // For UI testing purposes wait X seconds to see the loading state
+                await sleep(0);
```

Comment:
do you need this sleep? How is it working with a 0 sleep?

Reply:
Yes, for testing demo app loading state